opentracing-jdbc/opentracing-jdbc/src/main/java/org/zalando/opentracing/jdbc/TracingQueryExecutionListener.java

```diff
@@ -0,0 +1,61 @@
+package org.zalando.opentracing.jdbc;
+
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+import lombok.AllArgsConstructor;
+import lombok.SneakyThrows;
+import net.ttddyy.dsproxy.ExecutionInfo;
+import net.ttddyy.dsproxy.QueryInfo;
+import net.ttddyy.dsproxy.listener.QueryExecutionListener;
+import org.zalando.opentracing.jdbc.operation.OperationName;
+import org.zalando.opentracing.jdbc.span.SpanDecorator;
+
+import java.lang.reflect.Method;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+@AllArgsConstructor
+final class TracingQueryExecutionListener implements QueryExecutionListener {
+
+    private static final String SPAN = TracingQueryExecutionListener.class.getName() + ".SPAN";
+
+    private final Tracer tracer;
+    private final OperationName operationName;
+    private final SpanDecorator decorator;
+
+    @Override
+    @SneakyThrows(SQLException.class)
```

Comment:
I am afraid that this leads to undeclared checked exceptions coming by some Spring's `@Transactional` annotation.

Reply:
I believe it's an oversight in the datasource-proxy because the listener is executed within methods that allow SQLException. I opened https://github.com/ttddyy/datasource-proxy/issues/62